### PR TITLE
Bump controller-runtime to latest

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Master version of `controller-runtime` replacement in `go.mod` is refusing to build or vendor on my machine.
```
go: github.com/munnerz/controller-runtime@v0.1.8-0.20200318092001-e22ac1073450: invalid version: unknown revision e22ac1073450
```
As the changes needed were included in master, having this replacement makes no sense any more.

### CODE REVIEW

File: pkg/controller/machinehealthcheck/machinehealthcheck_controller.go [hunk 1]
```diff
@@ -80,7 +80,8 @@ func Add(mgr manager.Manager, opts manager.Options) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager, opts manager.Options) (*ReconcileMachineHealthCheck, error) {
-	if err := mgr.GetCache().IndexField(&mapiv1.Machine{},
+	if err := mgr.GetCache().IndexField(context.Background(),
```
Comment (JoelSpeed) [inline pkg/controller/machinehealthcheck/machinehealthcheck_controller.go:83]: I would consider putting this as a TODO so we remember to fix it up at some point
```suggestion
	if err := mgr.GetCache().IndexField(context.TODO(),
```

File: pkg/controller/nodelink/nodelink_controller.go [hunk 2]
```diff
@@ -117,28 +117,32 @@ func indexMachineByInternalIP(object runtime.Object) []string {
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) (*ReconcileNodeLink, error) {
 	// set convenient indexers
-	if err := mgr.GetCache().IndexField(&corev1.Node{},
+	if err := mgr.GetCache().IndexField(context.Background(),
```
Comment (JoelSpeed) [inline pkg/controller/nodelink/nodelink_controller.go:120]: Similarly, the ones in this file should probably be TODO, I imagine we might plumb a context through at some point

**Comment by enxebre** [comment]
@Danil-Grigorev could you please split the vendored commit from the code changes we did as a consequence?

**Comment by Danil-Grigorev** [comment]
> @Danil-Grigorev could you please split the vendored commit from the code changes we did as a consequence?

Just tried to do that - either tests will fail, or the code won't compile at all without changes in the way the `controller-runtime.Client` methods are called.

**Comment by enxebre** [comment]
Thanks @Danil-Grigorev!  for consistency with any other revendor: 
- can we please move the go.mod changes in to the revendor commit.
- can we rephrase "Bumped controller-runtime to latest" to use imperative? we set internal convention on https://chris.beams.io/posts/git-commit/#imperative e.g "Change controller to satisfy latest controller-runtime"
- Can we get the changes to the gen-crd script into its own commit?

**Comment by enxebre** [comment]
/hold
to make sure we address  before it gets merged https://github.com/openshift/machine-api-operator/pull/570#issuecomment-638077252

**Pull request merged**